packages/opentelemetry-exporter-zipkin/src/transform.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { ReadableSpan } from '@opentelemetry/basic-tracer';
+import * as zipkinTypes from './types';
+
+const MICROS_PER_MILLI = 1000;
+
+// TODO: what should be the mapping?
+const STATUS_CODE = 'ot.status_code';
+const STATUS_DESCRIPTION = 'ot.status_description';
+
+const ZIPKIN_SPAN_KIND_MAPPING = {
+  [types.SpanKind.CLIENT]: zipkinTypes.SpanKind.CLIENT,
+  [types.SpanKind.SERVER]: zipkinTypes.SpanKind.SERVER,
+  [types.SpanKind.CONSUMER]: zipkinTypes.SpanKind.CONSUMER,
+  [types.SpanKind.PRODUCER]: zipkinTypes.SpanKind.PRODUCER,
+  // TODO: discuss mapping
+  // Zipkin doesnt have type INTERNAL
+  [types.SpanKind.INTERNAL]: zipkinTypes.SpanKind.CLIENT,
+};
+
+/**
+ * Translate OpenTelemetry ReadableSpan to ZipkinSpan format
+ * @param span Span to be translated
+ */
+export function toZipkinSpan(
+  serviceName: string,
+  span: ReadableSpan
+): zipkinTypes.Span {
+  const zipkinSpan: zipkinTypes.Span = {
+    traceId: span.spanContext.traceId,
+    parentId: span.parentSpanId,
+    name: span.name,
+    id: span.spanContext.spanId,
+    kind: ZIPKIN_SPAN_KIND_MAPPING[span.kind],
+    timestamp: span.startTime * MICROS_PER_MILLI,
+    duration: (span.startTime - span.endTime) * MICROS_PER_MILLI,
```

Comment:
I think duration should be `(span.endTime - span.startTime)` and not `(span.startTime - span.endTime)`.

Reply:
I still think we should store the duration and not the end time. When I think about it, it doesn't even make sense to rely on the concept of an end time since it could end up before the start time because of daylight saving time. This makes the current implementation semantically incorrect.

Reply:
Regarding daylight savings time, my understanding is that `performance.now` is monotonic, so that wouldn't be an issue if we use it? Or am I missing something?

I'm open to storing the duration, but thought it could be simpler to just store start and end since that's what's provided either at measurement time or as a parameter